tools/src/bin/output-file.rs

```diff
@@ -24,6 +24,31 @@ use languages::FormatAs;
 
 use tools::output::{PanelItem, PanelSection};
 
+extern crate rustc_serialize;
+use rustc_serialize::json;
+
+fn read_json_from_file(path: &str) -> Option<json::Object> {
+    let components_file = File::open(path).ok()?;
+    let mut reader = BufReader::new(&components_file);
+    json::Json::from_reader(&mut reader).ok()?.into_object()
+}
+
+/// For a given path, looks up the bugzilla product and component and
+/// returns it in a tuple if it could be found. The JSON data format
+/// comes from https://searchfox.org/mozilla-central/rev/47edbd91c43db6229cf32d1fc4bae9b325b9e2d0/python/mozbuild/mozbuild/frontend/mach_commands.py#209-223,243
+/// and is fairly straightforward.
+fn get_bugzilla_component<'a>(bugzilla_data: &'a json::Object, path: &str) -> Option<(&'a str, &'a str)> {
+    let mut path_obj = bugzilla_data.get("paths")?;
+    for path_component in path.split('/') {
```

Comment:
Does this need to be different for windows? I guess since we're skipping generated files anyway it may not matter.

Reply:
No, these paths are all from the VM instance, so they will always have forward slashes (assuming that's what you meant by windows differences)